sqlite3/lib/src/impl/statement.dart

```diff
@@ -220,6 +272,16 @@ class PreparedStatementImpl implements PreparedStatement {
           parameters, 'parameters', 'Expected $count parameters, got $length');
     }
   }
+
+  void _ensureMatchingMapParameters(Map<String, Object?>? parameters) {
+    final length = parameters?.length ?? 0;
+    final count = parameterCount;
+
+    if (length != count) {
```

Comment:
I think this check is not enough, I think we should also check that

1. For each key in the map, a matching SQL variable exists.
2. Each SQL variable is bound to a value from the Dart map.

If we check 1., checking 2. is just a matter of comparing parameter counts but we need to check 1. I think the most efficient way to check this would be to integrate this into `_bindMapParams`.